### ISSUE TITLE
"java.lang.IllegalArgumentException: log10 p: Values must be non-infinite and non-NAN" in v4.1.4.0

### ISSUE THREAD

**Issue by winni2k** [broadinstitute/gatk#6255]
### Affected tool(s) or class(es)
GATK v4.1.4.0 using FilterMutectCalls

### Affected version(s)
- [x] Latest public release version `4.1.4.0` installed from conda release `gatk4-4.1.4.0-1`
- [ ] Latest master branch as of [date of test?]

### Description 
This issue reports the same error that is reported in #6237, but on the latest release, and in a mitochondrial calling setting.

My command is:
```bash
gatk FilterMutectCalls -V MT.vcf.gz\
       -R human_g1k_v37.main.fasta\
       -O MT.filtered.vcf.gz\
       --stats MT.vcf.gz.stats\
       --mitochondria-mode
```

I get the following output to STDERR:
```
11:15:57.152 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/warkre/miniconda3/envs/gatk4.1.4.0/share/gatk4-4.1.4.0-1/gatk-package-4.1.4.0-local.jar!/com/intel/gkl/native/libgkl_compression.so
Nov 07, 2019 11:15:57 AM shaded.cloud_nio.com.google.auth.oauth2.ComputeEngineCredentials runningOnComputeEngine
INFO: Failed to detect whether we are running on Google Compute Engine.
11:15:57.328 INFO  FilterMutectCalls - ------------------------------------------------------------
11:15:57.328 INFO  FilterMutectCalls - The Genome Analysis Toolkit (GATK) v4.1.4.0
11:15:57.328 INFO  FilterMutectCalls - For support and documentation go to https://software.broadinstitute.org/gatk/
11:15:57.328 INFO  FilterMutectCalls - Executing as warkre@fuji on Linux v4.9.0-9-amd64 amd64
11:15:57.328 INFO  FilterMutectCalls - Java runtime: OpenJDK 64-Bit Server VM v1.8.0_192-b01
11:15:57.329 INFO  FilterMutectCalls - Start Date/Time: November 7, 2019 11:15:57 AM CET
11:15:57.329 INFO  FilterMutectCalls - ------------------------------------------------------------
11:15:57.329 INFO  FilterMutectCalls - ------------------------------------------------------------
11:15:57.329 INFO  FilterMutectCalls - HTSJDK Version: 2.20.3
11:15:57.329 INFO  FilterMutectCalls - Picard Version: 2.21.1
11:15:57.329 INFO  FilterMutectCalls - HTSJDK Defaults.COMPRESSION_LEVEL : 2
11:15:57.329 INFO  FilterMutectCalls - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false
11:15:57.329 INFO  FilterMutectCalls - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true
11:15:57.329 INFO  FilterMutectCalls - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false
11:15:57.329 INFO  FilterMutectCalls - Deflater: IntelDeflater
11:15:57.329 INFO  FilterMutectCalls - Inflater: IntelInflater
11:15:57.329 INFO  FilterMutectCalls - GCS max retries/reopens: 20
11:15:57.329 INFO  FilterMutectCalls - Requester pays: disabled
11:15:57.329 INFO  FilterMutectCalls - Initializing engine
11:15:57.537 INFO  FeatureManager - Using codec VCFCodec to read file file:///tmp/tmp.8lRGFREUhm/MT.vcf.gz
11:15:57.553 INFO  FilterMutectCalls - Done initializing engine
11:15:57.599 INFO  ProgressMeter - Starting traversal
11:15:57.599 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute
11:15:57.599 INFO  FilterMutectCalls - Starting pass 0 through the variants
11:15:57.637 INFO  FilterMutectCalls - Finished pass 0 through the variants
11:15:57.657 INFO  FilterMutectCalls - Shutting down engine
[November 7, 2019 11:15:57 AM CET] org.broadinstitute.hellbender.tools.walkers.mutect.filtering.FilterMutectCalls done. Elapsed time: 0.01 minutes.
Runtime.totalMemory()=2148007936
java.lang.IllegalArgumentException: log10 p: Values must be non-infinite and non-NAN
	at org.broadinstitute.hellbender.utils.NaturalLogUtils.logSumExp(NaturalLogUtils.java:84)
	at org.broadinstitute.hellbender.utils.NaturalLogUtils.normalizeLog(NaturalLogUtils.java:51)
	at org.broadinstitute.hellbender.tools.walkers.mutect.clustering.SomaticClusteringModel.clusterProbabilities(SomaticClusteringModel.java:203)
	at org.broadinstitute.hellbender.tools.walkers.mutect.clustering.SomaticClusteringModel.learnAndClearAccumulatedData(SomaticClusteringModel.java:122)
	at org.broadinstitute.hellbender.tools.walkers.mutect.filtering.Mutect2FilteringEngine.learnParameters(Mutect2FilteringEngine.java:156)
	at org.broadinstitute.hellbender.tools.walkers.mutect.filtering.FilterMutectCalls.afterNthPass(FilterMutectCalls.java:157)
	at org.broadinstitute.hellbender.engine.MultiplePassVariantWalker.traverse(MultiplePassVariantWalker.java:44)
	at org.broadinstitute.hellbender.engine.GATKTool.doWork(GATKTool.java:1048)
	at org.broadinstitute.hellbender.cmdline.CommandLineProgram.runTool(CommandLineProgram.java:139)
	at org.broadinstitute.hellbender.cmdline.CommandLineProgram.instanceMainPostParseArgs(CommandLineProgram.java:191)
	at org.broadinstitute.hellbender.cmdline.CommandLineProgram.instanceMain(CommandLineProgram.java:210)
	at org.broadinstitute.hellbender.Main.runCommandLineProgram(Main.java:163)
	at org.broadinstitute.hellbender.Main.mainEntry(Main.java:206)
	at org.broadinstitute.hellbender.Main.main(Main.java:292)
/home/warkre/miniconda3/envs/gatk4.1.4.0/bin/gatk:80: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if len(args) is 0 or (len(args) is 1 and (args[0] == "--help" or args[0] == "-h")):
/home/warkre/miniconda3/envs/gatk4.1.4.0/bin/gatk:80: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if len(args) is 0 or (len(args) is 1 and (args[0] == "--help" or args[0] == "-h")):
/home/warkre/miniconda3/envs/gatk4.1.4.0/bin/gatk:117: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if len(args) is 1 and args[0] == "--list":
/home/warkre/miniconda3/envs/gatk4.1.4.0/bin/gatk:308: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if call(["gsutil", "-q", "stat", gcsjar]) is 0:
/home/warkre/miniconda3/envs/gatk4.1.4.0/bin/gatk:312: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if call(["gsutil", "cp", jar, gcsjar]) is 0:
/home/warkre/miniconda3/envs/gatk4.1.4.0/bin/gatk:467: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if not len(properties) is 0:
/home/warkre/miniconda3/envs/gatk4.1.4.0/bin/gatk:471: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if not len(filesToAdd) is 0:
Using GATK jar /home/warkre/miniconda3/envs/gatk4.1.4.0/share/gatk4-4.1.4.0-1/gatk-package-4.1.4.0-local.jar
Running:
    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /home/warkre/miniconda3/envs/gatk4.1.4.0/share/gatk4-4.1.4.0-1/gatk-package-4.1.4.0-local.jar FilterMutectCalls -V MT.vcf.gz -R human_g1k_v37.main.fasta -O MT.filtered.vcf.gz --stats MT.vcf.gz.stats --mitochondria-mode
```

#### Steps to reproduce

The data is sensitive human data, so I will not share that data. However, I have made up some data that appears to fail as well. Using the command:

```bash
gatk FilterMutectCalls -V mu.2.vcf         -R human_g1k_v37.main.fasta          -O MT.filtered.vcf.gz          --stats MT.vcf.gz.stats         --mitochondria-mode
```
I get an output to STDERR:
```
11:30:15.521 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/warkre/miniconda3/envs/gatk4.1.4.0/share/gatk4-4.1.4.0-1/gatk-package-4.1.4.0-local.jar!/com/intel/gkl/native/libgkl_compression.so
Nov 07, 2019 11:30:15 AM shaded.cloud_nio.com.google.auth.oauth2.ComputeEngineCredentials runningOnComputeEngine
INFO: Failed to detect whether we are running on Google Compute Engine.
11:30:15.673 INFO  FilterMutectCalls - ------------------------------------------------------------
11:30:15.674 INFO  FilterMutectCalls - The Genome Analysis Toolkit (GATK) v4.1.4.0
11:30:15.674 INFO  FilterMutectCalls - For support and documentation go to https://software.broadinstitute.org/gatk/
11:30:15.674 INFO  FilterMutectCalls - Executing as warkre@fuji on Linux v4.9.0-9-amd64 amd64
11:30:15.674 INFO  FilterMutectCalls - Java runtime: OpenJDK 64-Bit Server VM v1.8.0_192-b01
11:30:15.674 INFO  FilterMutectCalls - Start Date/Time: November 7, 2019 11:30:15 AM CET
11:30:15.674 INFO  FilterMutectCalls - ------------------------------------------------------------
11:30:15.674 INFO  FilterMutectCalls - ------------------------------------------------------------
11:30:15.674 INFO  FilterMutectCalls - HTSJDK Version: 2.20.3
11:30:15.674 INFO  FilterMutectCalls - Picard Version: 2.21.1
11:30:15.674 INFO  FilterMutectCalls - HTSJDK Defaults.COMPRESSION_LEVEL : 2
11:30:15.674 INFO  FilterMutectCalls - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false
11:30:15.674 INFO  FilterMutectCalls - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true
11:30:15.674 INFO  FilterMutectCalls - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false
11:30:15.674 INFO  FilterMutectCalls - Deflater: IntelDeflater
11:30:15.674 INFO  FilterMutectCalls - Inflater: IntelInflater
11:30:15.674 INFO  FilterMutectCalls - GCS max retries/reopens: 20
11:30:15.675 INFO  FilterMutectCalls - Requester pays: disabled
11:30:15.675 INFO  FilterMutectCalls - Initializing engine
11:30:15.870 INFO  FeatureManager - Using codec VCFCodec to read file file:///tmp/tmp.8lRGFREUhm/mu.2.vcf
11:30:15.883 INFO  FilterMutectCalls - Done initializing engine
11:30:15.929 INFO  ProgressMeter - Starting traversal
11:30:15.929 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute
11:30:15.930 INFO  FilterMutectCalls - Starting pass 0 through the variants
11:30:15.958 INFO  FilterMutectCalls - Finished pass 0 through the variants
11:30:15.970 INFO  FilterMutectCalls - Starting pass 1 through the variants
11:30:15.974 INFO  FilterMutectCalls - Shutting down engine
[November 7, 2019 11:30:15 AM CET] org.broadinstitute.hellbender.tools.walkers.mutect.filtering.FilterMutectCalls done. Elapsed time: 0.01 minutes.
Runtime.totalMemory()=2252865536
java.lang.IllegalArgumentException: log10 p: Values must be non-infinite and non-NAN
	at org.broadinstitute.hellbender.utils.NaturalLogUtils.logSumExp(NaturalLogUtils.java:84)
	at org.broadinstitute.hellbender.utils.NaturalLogUtils.normalizeLog(NaturalLogUtils.java:51)
	at org.broadinstitute.hellbender.tools.walkers.mutect.clustering.SomaticClusteringModel.clusterProbabilities(SomaticClusteringModel.java:203)
	at org.broadinstitute.hellbender.tools.walkers.mutect.clustering.SomaticClusteringModel.probabilityOfSequencingError(SomaticClusteringModel.java:96)
	at org.broadinstitute.hellbender.tools.walkers.mutect.filtering.TumorEvidenceFilter.calculateErrorProbability(TumorEvidenceFilter.java:27)
	at org.broadinstitute.hellbender.tools.walkers.mutect.filtering.Mutect2VariantFilter.errorProbability(Mutect2VariantFilter.java:15)
	at org.broadinstitute.hellbender.tools.walkers.mutect.filtering.ErrorProbabilities.lambda$new$1(ErrorProbabilities.java:19)
	at java.util.stream.Collectors.lambda$toMap$58(Collectors.java:1321)
	at java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169)
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1382)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
	at org.broadinstitute.hellbender.tools.walkers.mutect.filtering.ErrorProbabilities.<init>(ErrorProbabilities.java:19)
	at org.broadinstitute.hellbender.tools.walkers.mutect.filtering.Mutect2FilteringEngine.accumulateData(Mutect2FilteringEngine.java:141)
	at org.broadinstitute.hellbender.tools.walkers.mutect.filtering.FilterMutectCalls.nthPassApply(FilterMutectCalls.java:146)
	at org.broadinstitute.hellbender.engine.MultiplePassVariantWalker.lambda$traverse$0(MultiplePassVariantWalker.java:40)
	at org.broadinstitute.hellbender.engine.MultiplePassVariantWalker.lambda$traverseVariants$1(MultiplePassVariantWalker.java:77)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.Iterator.forEachRemaining(Iterator.java:116)
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
	at org.broadinstitute.hellbender.engine.MultiplePassVariantWalker.traverseVariants(MultiplePassVariantWalker.java:75)
	at org.broadinstitute.hellbender.engine.MultiplePassVariantWalker.traverse(MultiplePassVariantWalker.java:40)
	at org.broadinstitute.hellbender.engine.GATKTool.doWork(GATKTool.java:1048)
	at org.broadinstitute.hellbender.cmdline.CommandLineProgram.runTool(CommandLineProgram.java:139)
	at org.broadinstitute.hellbender.cmdline.CommandLineProgram.instanceMainPostParseArgs(CommandLineProgram.java:191)
	at org.broadinstitute.hellbender.cmdline.CommandLineProgram.instanceMain(CommandLineProgram.java:210)
	at org.broadinstitute.hellbender.Main.runCommandLineProgram(Main.java:163)
	at org.broadinstitute.hellbender.Main.mainEntry(Main.java:206)
	at org.broadinstitute.hellbender.Main.main(Main.java:292)
/home/warkre/miniconda3/envs/gatk4.1.4.0/bin/gatk:80: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if len(args) is 0 or (len(args) is 1 and (args[0] == "--help" or args[0] == "-h")):
/home/warkre/miniconda3/envs/gatk4.1.4.0/bin/gatk:80: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if len(args) is 0 or (len(args) is 1 and (args[0] == "--help" or args[0] == "-h")):
/home/warkre/miniconda3/envs/gatk4.1.4.0/bin/gatk:117: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if len(args) is 1 and args[0] == "--list":
/home/warkre/miniconda3/envs/gatk4.1.4.0/bin/gatk:308: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if call(["gsutil", "-q", "stat", gcsjar]) is 0:
/home/warkre/miniconda3/envs/gatk4.1.4.0/bin/gatk:312: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if call(["gsutil", "cp", jar, gcsjar]) is 0:
/home/warkre/miniconda3/envs/gatk4.1.4.0/bin/gatk:467: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if not len(properties) is 0:
/home/warkre/miniconda3/envs/gatk4.1.4.0/bin/gatk:471: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if not len(filesToAdd) is 0:
Using GATK jar /home/warkre/miniconda3/envs/gatk4.1.4.0/share/gatk4-4.1.4.0-1/gatk-package-4.1.4.0-local.jar
Running:
    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /home/warkre/miniconda3/envs/gatk4.1.4.0/share/gatk4-4.1.4.0-1/gatk-package-4.1.4.0-local.jar FilterMutectCalls -V mu.2.vcf -R human_g1k_v37.main.fasta -O MT.filtered.vcf.gz --stats MT.vcf.gz.stats --mitochondria-mode
```

The made-up VCF contains a single variant:
```
MT      100     .       G       C       .       .       DP=3;ECNT=3;MBQ=0,10;MFRL=0,10;MMQ=10,10;MPOS=10;OCM=0;POPAF=3.40;RPA=5,6;RU=C;STR;TLOD=5.88    GT:AD:AF:DP:F1R2:F2R1:SB        0/1:0,2:0.750:2:0,1:0,1:0,0,0,2
```

The stats file contains the lines:
```
statistic	value
callable	0.0
```

I hope this is enough to reproduce the error.

**Comment by davidbenjamin** [comment]
@winni2k If this hasn't already been fixed then PR #6337 should resolve it once merged.

**Comment by tmm211** [comment]
[Another user](https://gatk.broadinstitute.org/hc/en-us/community/posts/360057968772-FilterMutectCalls-Error-java-lang-IllegalArgumentException-log10-p-Values-must-be-non-infinite-and-non-NAN) reported seeing this same issue for 4.1.4.1.

**Comment by davidbenjamin** [comment]
We can reopen if anyone reports the issue in 4.1.5.0.

**Comment by davidbenjamin** [comment]
Hi Tiffany,

That's fine because the fix wasn't until 4.1.5.0.

**Comment by tmm211** [comment]
All good! The user said the issue went away with 4.1.5.0
